DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/ViewTreeRecorder.swift

```diff
@@ -36,7 +36,17 @@ internal struct ViewTreeRecorder {
             context.viewControllerContext.isRootView = false
         }
 
-        let semantics = nodeSemantics(for: view, in: context)
+        // Convert the frame in root view space
+        let frame = view.convert(view.bounds, to: context.coordinateSpace)
+
+        if view.clipsToBounds {
+            // Propagate view's clipping intersection when clipsToBounds is
+            // enabled.
+            context.clip = frame.intersection(context.clip)
+        }
+
+        let attributes = ViewAttributes(view: view, frame: frame, clip: context.clip)
+        let semantics = nodeSemantics(for: view, with: attributes, in: context)
```

Comment:
Here, we propagate the `clip` rectangle by intersection the view frame with the current `clip`, only when `clipsToBounds` is enabled.